### ISSUE TITLE
Fail to to enable server-side keepAlive on 1.3.0 RELEASE

### ISSUE THREAD

**Issue by bobwenx** [grpc/grpc-java#2982]
### What version of gRPC are you using?
GRPC 1.3.0

### What JVM are you using (`java -version`)?
1.8.0_112

### What did you do?
Try to enable server-side keepAlive(added in gprc-java 1.3) but the feature didn't work properly.
After setting server-side keepAlive to 180s and keepAliveTimeout to 30s. what i expect is the server will sending a keepAlive to client if the connection is idle(didn't sent any data during 180s). however. i did't see such ping packet on network. 

Debugging the code(io.grpc.internal.KeepAliveManager), it seems the `onTransportActive`/`onTransportIdle` methods didn't be called by NettyServerHandler.  Unlike the NettyClientHandler, the keepAliveManager can't be triggered by stream's created/remove.

Only `onDataReceived` is called if data received by NettyServerHandler. 
![image](https://cloud.githubusercontent.com/assets/2533748/25769933/895356b2-3259-11e7-90c3-84a064ad656c.png)

So, simply if the client didn't sent any data, the server wouldn't sent back any ping.

In this case, another way to enable server-side KeepAlive is set `keepAliveDuringTransportIdle(permitKeepAliveWithoutCalls)` to true, so when the `io.grpc.internal.KeepAliveManager#onTransportStarted` is called, it will schedule a ping future immediately. 

However, the call of `io.grpc.internal.KeepAliveManager#onTransportStarted` didn't make effective too!
the only caller on server is `io.grpc.netty.NettyServerHandler#handleProtocolNegotiationCompleted`, which didn't invoke `onTransportStarted` because the `keepAliveManager` has not been initialized yet(the keepAliveManager always null during this method called).

![image](https://cloud.githubusercontent.com/assets/2533748/25769901/78056e0a-3258-11e7-9a20-47baea059d6c.png)

**Comment by bobwenx** [comment]
BTW, i can confirm the client-side keepAlive is worked right now :)

**Comment by ejona86** [comment]
> Unlike the NettyClientHandler, the keepAliveManager can't be triggered by stream's created/remove.

This should be fine, because it is created with "keepAliveDuringTransportIdle" (the last boolean):

```java
keepAliveManager = new KeepAliveManager(new KeepAlivePinger(ctx), ctx.executor(),
    keepAliveTimeInNanos, keepAliveTimeoutInNanos, true /* keepAliveDuringTransportIdle */);
```

> However, the call of io.grpc.internal.KeepAliveManager#onTransportStarted didn't make effective too!
the only caller on server is io.grpc.netty.NettyServerHandler#handleProtocolNegotiationCompleted, which didn't invoke onTransportStarted because the keepAliveManager has not been initialized yet(the keepAliveManager always null during this method called).

Ah, that looks much more like the problem. I am pretty sure I manually tested it, but I would have only tested a single negotiator.

@bobwenx, did you notice the problem with TLS or plaintext? I think I may have tested TLS, but I'll check them both again.

**Comment by bobwenx** [comment]
i only test it with plaintext negotiator

**Comment by ejona86** [comment]
Hmm... Actually, I guess I didn't test this since @dapengzhang0 was working on it. Looking at #2987 it seems maybe it doesn't work at all today (:: sigh ::).

**Comment by dapengzhang0** [comment]
@bobwenx  thanks a lot for testing the new API and reporting issues.

The problem was `grpcHandler.handleProtocolNegotiationCompleted` can happen before `ctx.pipeline().replace(this, null, grpcHandler)` in which `handlerAdded()` is called

Moved `keepAliveManager.onTransportStarted()` inside `handlerAdded()` to fix it in  #2987.
